File: src/main/java/com/example/parking/domain/reservation/dto/ReservationResDto.java
package com.example.parking.domain.reservation.dto;

import com.example.parking.domain.reservation.entity.Reservation;
import com.example.parking.domain.reservation.entity.ReservationStatus;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

public record ReservationResDto(
        Long reservationId,
        Long parkingLotId,        // 추가
        Long parkingSpotId,       // 추가
        String parkingLotName,
        String parkingSpotNumber,
        LocalDateTime startTime,
        LocalDateTime endTime,
        ReservationStatus status,
        Integer totalPrice        // 추가
) {
    public static ReservationResDto from(Reservation reservation) {
        long minutes = ChronoUnit.MINUTES.between(
                reservation.getStartTime(), reservation.getEndTime()
        );
        int price = (int) Math.ceil(minutes / 10.0) * reservation.getParkingLot().getPrice();

        return new ReservationResDto(
                reservation.getId(),
                reservation.getParkingLot().getId(),       // 추가
                reservation.getParkingSpot().getId(),      // 추가
                reservation.getParkingLot().getName(),
                reservation.getParkingSpot().getNumber(),
                reservation.getStartTime(),
                reservation.getEndTime(),
                reservation.getStatus(),
                price                                      // 추가
        );
    }
}

File: src/main/java/com/example/parking/domain/reservation/dto/ReservationResDto.kt
package com.example.parking.domain.reservation.dto

import com.example.parking.domain.reservation.entity.Reservation
import com.example.parking.domain.reservation.entity.ReservationStatus
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit

data class ReservationResDto(
    val reservationId: Long?,
    val parkingLotId: Long?,
    val parkingSpotId: Long?,
    val parkingLotName: String,
    val parkingSpotNumber: String,
    val startTime: LocalDateTime,
    val endTime: LocalDateTime,
    val status: ReservationStatus,
    val totalPrice: Int
) {
    companion object {
        @JvmStatic
        fun from(reservation: Reservation): ReservationResDto {
            val minutes = ChronoUnit.MINUTES.between(reservation.startTime, reservation.endTime)
            val price = (Math.ceil(minutes / 10.0) * reservation.parkingLot.price).toInt()

            return ReservationResDto(
                reservationId = reservation.id,
                parkingLotId = reservation.parkingLot.id,
                parkingSpotId = reservation.parkingSpot.id,
                parkingLotName = reservation.parkingLot.name,
                parkingSpotNumber = reservation.parkingSpot.number,
                startTime = reservation.startTime,
                endTime = reservation.endTime,
                status = reservation.status,
                totalPrice = price
            )
        }
    }
}